crates/base/src/datetimes.rs

```diff
@@ -71,15 +77,30 @@ impl FromStr for TimeZoneId {
     type Err = BaseError;
 
     fn from_str(tz_name: &str) -> BaseResult<TimeZoneId> {
```

Comment:
for the from_str and further tz string included logic, it is better like this: all string is convert to offset in seconds. We do not maintain two internal id form and not any trick necessary. That's, we discard DST support although there is a room for future.